Add pre-commit checks

keirthana opened the pull request:
* Add a pre-commit hook for inclusive naming check
* Fix issues reported by the pre-commit hook

keirthana committed:
Add pre-commit inclusive naming check

keirthana committed:
Fix issues reported by woke

keirthana commented on README.rst:
```diff
@@ -127,3 +127,22 @@ Some notes about translating the guide:
+3. Navigate to the top-level directory of this repository and run ``pre-commit install --install-hooks``.
+4. The following checks are run automatically before every commit:
+  - Inclusive naming checks (``woke``).
+5. To ignore any violation reported by ``woke``, see `<https://docs.getwoke.tech/ignore/>`_.
```
I understand, the intention behind linking to the documentation was so that we don't have outdated instructions in case the woke decides to change something. I will add some preliminary ignoring instructions and then maybe redirect for more information.

s-makin replied:
Sounds good to me! Thanks :)

keirthana committed:
Add examples for woke ignore

s-makin reviewed (approved):
LGTM! Thanks for adding this. Will let Dominik review/merge to make sure I haven't overlooked anything.

dviererbe commented on README.rst:
```diff
@@ -127,3 +127,34 @@ Some notes about translating the guide:
+
+Optionally, you can run pre-commit checks on your changes. To do so:
+
+1. Install the ``pre-commit`` package from the Ubuntu repositories::
+
+    sudo apt install pre-commit
+
+2. Install the ``woke`` snap by running::
+
+    sudo snap install woke
+
+   Alternatively, you can install it by following instructions in `<https://docs.getwoke.tech/installation/>`_.
```
Question: Do we want to add the installation of `woke` and `pre-commit` to the Makefile?

keirthana replied:
@dviererbe I have added this after refactoring the Makefile to avoid duplication.

dviererbe commented on README.rst:
```diff
@@ -127,3 +127,34 @@ Some notes about translating the guide:
+  - Inclusive naming checks (``woke``).
+5. You can ignore a violation reported by ``woke``.
+  - To ignore an entire file, specify the file name in the ``.wokeignore`` file at the root of this repository.
+    You can create it if it doesn't exist already::
```
Suggestion: Can we just create an empty `.wokeignore` file and remove this line?

keirthana replied:
@dviererbe Done, I had to add a pattern to the file anyway

dviererbe reviewed (approved):
Praise: Nice addition, LGTM
Note: I will merge this once my 2 comments are addressed.

keirthana committed:
Address review comments

dviererbe committed:
Add `_build` directory to `.wokeignore` file

This is technically not needed as the `.pre-commit-config.yaml` file
specifies that woke only checks `.rst` files, but this may avoid
unwanted warnings/checks in the future if the config should chnage in
the future.
